validation/immersed_boundaries/immersed_lat_lon_barotropic.jl

```diff
@@ -0,0 +1,237 @@
+# # Freely decaying barotropic turbulence on a latitude-longitude strip
+
+using Oceananigans
+using Oceananigans.Grids
+
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+
+using Oceananigans.Coriolis: HydrostaticSphericalCoriolis
+using Oceananigans.ImmersedBoundaries: ImmersedBoundaryGrid, GridFittedBoundary
+
+using Oceananigans.Models.HydrostaticFreeSurfaceModels:
+    HydrostaticFreeSurfaceModel,
+    VerticalVorticityField,
+    VectorInvariant,
+    ExplicitFreeSurface,
+    ImplicitFreeSurface
+
+using Oceananigans.TurbulenceClosures:
+    HorizontallyCurvilinearAnisotropicDiffusivity,
+    HorizontallyCurvilinearAnisotropicBiharmonicDiffusivity
+
+using Oceananigans.Utils: prettytime, hours, day, days, years, year
+using Oceananigans.OutputWriters: JLD2OutputWriter, TimeInterval, IterationInterval
+using Oceananigans.AbstractOperations: KernelFunctionOperation
+
+using Statistics
+using JLD2
+using Printf
+
+import Oceananigans.Architectures: architecture
+architecture(ibg::ImmersedBoundaryGrid) = architecture(ibg.grid)
+
+#####
+##### Grid
+#####
+
+latitude = (-30, 30)
+Δφ = latitude[2] - latitude[1]
+
+resolution = 4 # degree
+Nx = round(Int, 360 / resolution)
+Ny = round(Int, Δφ / resolution)
+Nz = 10
+
+# A spherical domain
+underlying_grid = RegularLatitudeLongitudeGrid(size = (Nx, Ny, Nz),
```

Comment:
Looks like there are a few points in z, so its just the name of the PR that has to change!